openerp/addons/base/ir/ir_attachment.py

```diff
@@ -141,6 +141,16 @@ def _get_path(self, cr, uid, bin_data, sha):
             os.makedirs(dirname)
         return fname, full_path
 
+    def _check(self, cr, uid, fname):
+        """ Add ``fname`` in a checklist for the filestore garbage collection. """
+        # add an empty file like "check/fname" in the filestore
+        full_path = self._full_path(cr, uid, "check/" + fname)
+        if not os.path.exists(full_path):
+            dirname = os.path.dirname(full_path)
+            if not os.path.isdir(dirname):
+                os.makedirs(dirname)
```

Comment:
There's a race condition here, two processes could `_check` at the same time, find out the directory doesn't already exist at the same time, and one of them will lose the race and blow up. This needs to handle `OSError(errno.EEXIST)`.